backtester/funding/funding.go

```diff
@@ -52,34 +71,67 @@ func CreateItem(exch string, a asset.Item, ci currency.Code, initialFunds, trans
 	}
 
 	return &Item{
-		exchange:     exch,
+		exchange:     strings.ToLower(exch),
 		asset:        a,
 		currency:     ci,
 		initialFunds: initialFunds,
 		available:    initialFunds,
 		transferFee:  transferFee,
-		snapshot:     make(map[time.Time]ItemSnapshot),
+		snapshot:     make(map[int64]ItemSnapshot),
 	}, nil
 }
 
+// LinkCollateralCurrency links an item to an existing currency code
+// for collateral purposes
+func (f *FundManager) LinkCollateralCurrency(item *Item, code currency.Code) error {
+	if !item.asset.IsFutures() {
+		return errNotFutures
+	}
+	for i := range f.items {
+		if f.items[i].currency.Equal(code) && f.items[i].asset == item.asset {
+			item.pairedWith = f.items[i]
+			return nil
+		}
+	}
+	collateral := &Item{
+		exchange:     item.exchange,
+		asset:        item.asset,
+		currency:     code,
+		pairedWith:   item,
+		isCollateral: true,
+	}
+	if err := f.AddItem(collateral); err != nil {
+		return err
+	}
+	item.pairedWith = collateral
+	return nil
+}
+
 // CreateSnapshot creates a Snapshot for an event's point in time
 // as funding.snapshots is a map, it allows for the last event
 // in the chronological list to establish the canon at X time
 func (f *FundManager) CreateSnapshot(t time.Time) {
 	for i := range f.items {
 		if f.items[i].snapshot == nil {
-			f.items[i].snapshot = make(map[time.Time]ItemSnapshot)
+			f.items[i].snapshot = make(map[int64]ItemSnapshot)
+			if f.items[i].isCollateral {
+				f.items[i].initialFunds = f.items[i].available
+			}
+		} else if _, ok := f.items[i].snapshot[t.Unix()]; ok {
```

Comment:
Question: How come we are using unix instead of unix nano?

Reply:
I was thinking in terms of large candles. UnixNano is better, especially for custom data